.github/scripts/ensure_actions_will_cancel.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+import argparse
+import sys
+import yaml
+
+from pathlib import Path
+
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+WORKFLOWS = REPO_ROOT / ".github" / "workflows"
+
+
+def concurrency_key(filename):
+    workflow_name = filename.with_suffix("").name.replace("_", "-")
+    return f"{workflow_name}-${{{{ github.event.pull_request.number || github.sha }}}}"
+
+
+SKIP = {
```

Comment:
Will this list have to be updated in the future/how do we plan on letting future authors of workflows know this?

Reply:
Oh actually yeah I have the same question; our current `regenerate_cancel_redundant_workflow.py` script just used this logic:

https://github.com/pytorch/pytorch/blob/17fb651a3bda74dfa5ad5437af8b7f622acecb37/.github/scripts/regenerate_cancel_redundant_workflow.py#L27

I see that the same logic is present in the `should_skip` function below, so is this `SKIP` variable actually necessary?

Reply:
Ah I guess they'll find out when they fail the quickcheck

Reply:
Copied over the logic from there, the check is a lot simpler now and there's no list to maintain